homeassistant/components/websocket_api/automation.py

```diff
@@ -138,6 +150,42 @@ def _get_automation_component_domains(
     return domains
 
 
+def _get_automation_component_lookup_table(
+    hass: HomeAssistant, component_descriptions: Mapping[str, Mapping[str, Any] | None]
+) -> _AutomationComponentLookupTable:
+    """Get a dict of automation components keyed by domain, along with the total number of components."""
+
+    if AUTOMATION_COMPONENT_LOOKUP_CACHE not in hass.data:
+        hass.data[AUTOMATION_COMPONENT_LOOKUP_CACHE] = []
```

Comment:
I know this is copying an existing pattern, but why don't we just create the cache when setting up the websocket handlers so we don't need to check if we have this key each time?

As an alternative, do this (`try` is basically free in recent Python releases, you only pay for exception handling and that is guaranteed to only happen once in the lifetime of the program):

```py
try:
    cache = hass.data[AUTOMATION_COMPONENT_LOOKUP_CACHE]
except KeyError;
    cache = hass.data[AUTOMATION_COMPONENT_LOOKUP_CACHE] = []
```

We should improve the other caches with this pattern in the same manner.